Assignment 3

omaralfaqih6 opened the pull request:
<!-- PULL REQUEST TITLE: UofT-DSI | <Module Name> - Assignment <assignment number>-->

## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)

## What did you learn from the changes you have made?

## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?

## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?

## How were these changes tested?

## A reference to a related issue in your repository (if applicable)

## Checklist
- [ ] I can confirm that my changes are working as intended

omaralfaqih6 committed:
I completed the assignment, but still the numeric_predictors variable is not as expected. Work in progress

omaralfaqih6 committed:
I reviewed and corrected the assignment, waiting on Julia to get back to us on the predictors variable in Q3 step 2.

omaralfaqih6 committed:
I completed the assignment and confirmed the fix of the Q3 Step2 coefficient_df parameters by replace lm with mlm.

omaralfaqih6 committed:
I reviewed and removed unnecessary lines and comments.

omaralfaqih6 committed:
Assignment is completed and I updated Q3 step 2 to use lm() rather than mlm() as stated in the question.

omaralfaqih6 committed:
I completed Q1-4, waiting on next week's lecture on bootstrapping to solve Q5.

omaralfaqih6 committed:
I updated Q2 on the importance of standardizing the data.

omaralfaqih6 committed:
I completed Q5, but Q3&4 needs to be revised.

omaralfaqih6 committed:
I completed answering Q5 and the three descriptive questions at the end.

omaralfaqih6 committed:
I fixed the answer of the last question in Q5.

omaralfaqih6 commented:
Please ignore assignment 2 file as I updated it by mistake. Assignment 2 file is done in assignment 2 branch.

PatelVishakh reviewed (changes requested):
Assignment 3 incomplete

Q2) Data Visualization

Not Looking for Linear relationships when seeing if clustering is applicable. 

Identify graphs that exhibit distinct clusteres.

Q5) Need to use the full size for bootstrapping, while you use 40 here. Also your confidence interval is using the replicate column when computing the conf thus the issue.

omaralfaqih6 committed:
I addressed both comments posted by Vishakh Patel on Q2 & Q5. The assignment is ready for a second review.

omaralfaqih6 commented:
@PatelVishakh thanks for the feedback! Assignment Ready for review!
Now I addressed your first point on Q2. Please check!
Regarding Q5, I was assuming that the bootstrapping was supposed to take a limited sample not the whole set, and I asked the learning staff during office hours and they advised to use number 40 for the sampling, but now i corrected it. Regarding the confidence interval, I fixed the issue, please check.
Regarding the confidence interval, I think i missed your point, I'm using ci_mean column to calculate it, can you elaborate more on the use of replicate column comment?

PatelVishakh commented:
Hi @omaralfaqih6,

Assignment 3 Complete.

Yes, that n= 40 approach was taken in Learning session but only as a demonstration. In practise, unless we are working with excessive amounts of data (ie big data regime) we usually let our bootstrap sample size be the same as the original sample size. 

Regarding the Conf. Int comment, I miss read your output. Although your confidence interval computed using (2.5%, 92.5%) is a valid conf. int. usually in practice we use (5%, 95%). This is so that we equally weight the datapoints in the tails on both sides (0% to 5% and 95% to 100%). Whereas your interval is going to be biased to the data points in the left tail of the distribution (2.5% to 5%) and leave out data points in the right tail (92.5% to 95%). This shift in quantiles will cause your conf. interval to wider then needed.

omaralfaqih6 commented:
@PatelVishakh thank you so much for the clarification. One point, before I close the PR, I had another review to the code and I updated it per your comments 5%-95% instead of 2.5% - 92.5%. Also, regarding the use of replicate column, I removed it and I calculated the mean of samples and converted them into a Numpy array.

Would you please have another look into the code - just for the sake of learning.

I really appreciate your feedback on the update code!
Thanks again.

omaralfaqih6 committed:
I revamped Q5 response confidence interval to be 5%-95% instead of 2.5%-92.5% & I removed the replicate.

omaralfaqih6 committed:
I added comments in Q5 for clarifty.
